### PR TITLE
Use EnergyCardBaseConfig where appropriate

### PR DESCRIPTION
**Pull request by TurnrDev**
## Breaking change
Hopefully not?

## Proposed change
Use `EnergyCardBaseConfig` as the new base for the Energy Cards which have collection_key

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

No changes to configuration

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **Refactor**
  - Improved consistency and maintainability by standardizing energy card configurations to extend from a unified `EnergyCardBaseConfig`.
  - ~~Enhanced energy device graph cards with a new optional `max_devices` property, allowing users to limit the number of devices displayed.~~ (CodeRabbit is drunk.)

### CODE REVIEW

**Comment by TurnrDev** [comment]
Found another one with an incorrect key too

**Comment by TurnrDev** [comment]
@coderabbitai review

**Comment by TurnrDev** [comment]
@coderabbitai review

File: src/panels/lovelace/cards/types.ts
Comment (TurnrDev) [inline src/panels/lovelace/cards/types.ts:1]: @coderabbitai run prettier to resolve linting for me

Comment (TurnrDev) [reply src/panels/lovelace/cards/types.ts:1]: I'm lazy and edited via the GitHub UI

File: src/panels/lovelace/cards/types.ts [hunk 1]
```diff
@@ -118,85 +118,72 @@ export interface EnergyCardBaseConfig extends LovelaceCardConfig {
...
-  extends LovelaceCardConfig {
+  extends EnergyCardBaseConfig {
   type: "energy-self-sufficiency-gauge";
   title?: string;
-  collection_key?: string;
 }
 
-export interface EnergyGridGaugeCardConfig extends LovelaceCardConfig {
-  type: "energy-grid-result-gauge";
+export interface EnergyGridGaugeCardConfig extends EnergyCardBaseConfig {
```
Comment (bramkragten) [inline src/panels/lovelace/cards/types.ts:179]: Can we rename this type too?

Comment (TurnrDev) [reply src/panels/lovelace/cards/types.ts]: Sorry it took me a while I didn't have the repo locally

**Comment by TurnrDev** [comment]
@coderabbitai you're seeing things that didn't change

**Review by bramkragten — APPROVED** [review]

**Pull request merged**